### PR TITLE
Validator displays name fields from error path

### PR DESCRIPTION
**Pull request by b-wils**

### CODE REVIEW

**Comment by mbasaglia** [comment]
I'm not too sure about this:

* The code style doesn't match the rest of `validator.js`
* It's display code so it should probably go into `validator/index.md` (`validator.js` is what will become a stand-alone library)
* If the JSON is missing `mn` in one of the objects in the path, the path becomes a bit misleading

**Comment by b-wils** [comment]
* Is there a style guide or summary somewhere? I can take another pass to try to infer and match, but it's quite a bit different from what we use.
* That's a good point, the joining is more display but I think the extraction into an array would be be usable. I can make that change.
* I'm not sure I follow the `mn` concern. Would adding a default value in cases where it is missing help fix that?

**Comment by mbasaglia** [comment]
Regarding `mn`, if you have layer with `mn: Layer1`, then a group shape without `mn`, and a shape with `mn: Ellipse`, it will show `Layer1 > Ellipse` which is a bit misleading since there one more step between them

**Comment by b-wils** [comment]
Addressed the concerns. Style should be closer, but let me know if there are still mismatches.

File: docs/static/js/validator.js [hunk 1]
```diff
@@ -462,7 +462,14 @@ class Validator
 
         let errors = [];
         if ( !this._validate_internal(data) )
-            errors = this._validate_internal.errors.map(e => this._cleaned_error(e));
+            errors = this._validate_internal.errors
+                .map(e => this._cleaned_error(e))
+                .map(e => {
+                    return {
+                        ...e,
+                        path_names: this._get_friendly_path_names(e.path, data)
```
Comment (mbasaglia) [inline docs/static/js/validator.js:470]: Might be worth adding to _cleaned_error to avoid a second pass with `map()`

Comment (b-wils) [reply docs/static/js/validator.js]: Done

**Review by mbasaglia — APPROVED** [review]

**Pull request merged**